test/cpp/api/dataloader.cpp

```diff
@@ -616,8 +616,8 @@ struct UnCopyableDataset : public datasets::Dataset<UnCopyableDataset> {
   ~UnCopyableDataset() = default;
 
   Example<> get(size_t index) override {
-    return {torch::tensor(static_cast<int64_t>(index)),
-            torch::tensor(static_cast<int64_t>(index))};
+    return {torch::tensor({static_cast<int64_t>(index)}),
+            torch::tensor({static_cast<int64_t>(index)})};
```

Comment:
I changed these tensors from 0-D to 1-D, so that the original behavior of the tests are preserved.